src/Bootstrappers/UrlTenancyBootstrapper.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Bootstrappers;
+
+use Closure;
+use Illuminate\Contracts\Routing\UrlGenerator;
+use Stancl\Tenancy\Contracts\TenancyBootstrapper;
+use Stancl\Tenancy\Contracts\Tenant;
+
+class UrlTenancyBootstrapper implements TenancyBootstrapper
+{
+    public static Closure|null $rootUrlOverride = null;
+    protected string|null $originalRootUrl = null;
+
+    public function __construct(protected UrlGenerator $urlGenerator)
+    {
+    }
```

Comment:
```suggestion
    public function __construct(
        protected UrlGenerator $urlGenerator,
    ) {}
```
We put the parameters on separate lines when using DI using parameters

Reply:
Right, I keep forgetting this. Though multiple lines feel natural when there are multiple parameters

Reply:
It's not just about multiple parameters. It's so that it's clear that we're using this for **creating the properties as well**, and not just injecting dependencies. The `protected` part can be easy to miss when it's all on one line.